homework/index.js

```diff
@@ -30,18 +32,129 @@
     return elem;
   }
 
-  function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
-      } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
-      }
+  function createContainer(selectedRepository) {
+    const container = document.getElementById('container');
+    const leftDiv = createAndAppend('div', container, {
+      id: 'leftDiv',
+      class: 'left-div white_Frame',
+    });
+    const table = createAndAppend('table', leftDiv);
+    const tbody = createAndAppend('tbody', table, { id: 'tbody' });
+
+    if (selectedRepository.name !== null) {
+      const tr1 = createAndAppend('tr', tbody);
+      createAndAppend('td', tr1, { class: 'label', text: 'Repository :' });
+      const td2 = createAndAppend('td', tr1);
+      createAndAppend('a', td2, {
+        href: selectedRepository.html_url,
+        target: '_blank',
+        text: selectedRepository.name,
+      });
+    }
```

Comment:
All the code from here on down essentially, could be refactored into a function that generates a role.  Each of the label and value are just strings.
If you want to add a detector that the `value` parameter is an object and not `null` you can use the function to generate the first row as well. 😄

Reply:
I should have given an example here, sorry:
```javascript
 const newRow = (parentTable, label, content) => {
    const row = createAndAppend('tr', parentTable);
    createAndAppend('td', row, { text: label, class: 'label' });
    // insert content
    if (typeof content !== 'object' || content === null) {
      createAndAppend('td', row, { text: `${!content ? '' : content}` });
    } else {
      const cell = createAndAppend('td', row);
      cell.appendChild(content);
    }
    return row;
  };
```
You can see here that we check if the content is not an `object` or is `null` (because `null` is considered an object in javascript) and then we know it is something that can be put in a string template, so we shove it in. 😄   Otherwise we are assuming it is a collection of HTML objects that just need to be appended to the parent.

Reply:
Sorry @rohni  I had got wrong you. Should I delete function that `detectRepositoryAndValue()` that checking if  the repository and repository's property and value are exist?